tools/get_git_commit.sh

```diff
@@ -5,19 +5,20 @@ cd ..
 output="./include/vrv/git_commit.h"
 COMMIT=""
 
-if [ ! -d "./.git" ]; then
-     echo "This is not a git directory and the git commit sha will remain undefined"
-else
+if git rev-parse --is-inside-work-tree >/dev/null 2>&1; then
     SHA=$(git describe  --exclude '*' --abbrev=7 --always --dirty)
     if [ -z "$SHA" ]; then
         echo "Undefined git commit version"
     else
         COMMIT="-$SHA"
     fi
+else
+    echo "This is not a git directory and the git commit sha will remain undefined"
 fi
 
 echo "////////////////////////////////////////////////////////" > $output
 echo "/// Git commit version file generated at compilation ///" >> $output
+echo "/// Timestamp: $(date +"%Y-%m-%dT%H:%M:%S%:z")             ///" >> $output
```

Comment:
```suggestion
echo "/// Timestamp: $(date -u +"%Y-%m-%dT%H:%M:%SZ")             ///" >> $output
```

`%:z` is not correct, since this outputs a literal `:z`. `%Z` will output the timezone (CEST); `date -u` will output the time in UTC. 

Just the letter `Z` at the end will print the literal, which is Zulu time, or the same as UTC. So:

`date -u +"%Y-%m-%dT%H:%M:%SZ"` -> `2025-04-22T11:39:53Z`
`date -u +"%Y-%m-%dT%H:%M:%S%Z" -> `2025-04-22T11:39:53UTC`

I think the suggested format is what is intended.

Reply:
My idea was to use ISO 8601 date format (see [this comment](https://unix.stackexchange.com/a/629504)) but using UTC is also fine for me.

Reply:
The SO post said that the date command on macOS doesn’t support that